build/checkdeps.sh

```diff
@@ -14,11 +14,22 @@ echo "checking that 'vendor' matches manifest"
 
 echo "checking that all deps are in 'vendor''"
 
-missing=$(sed -n 's,[[:space:]]*_[[:space:]]*"\(.*\)",\1,p' build/tool_imports.go | awk '{print} END {print "./pkg/..."}' \
-  | xargs go list -f '{{ join .Deps "\n"}}{{"\n"}}{{ join .TestImports "\n"}}{{"\n"}}{{ join .XTestImports "\n"}}' \
-  | grep -v '^github.com/cockroachdb/cockroach' \
-  | sort | uniq \
-  | xargs go list -f '{{if not .Standard}}{{.ImportPath}}{{end}}')
+top_deps=$(go list -f '{{join .Imports "\n"}}{{"\n"}}{{join .TestImports "\n"}}{{"\n"}}{{join .XTestImports "\n"}}' ./pkg/...)
+cmd_deps=$(sed -n 's,[[:space:]]*_[[:space:]]*"\(.*\)",./vendor/\1,p' build/tool_imports.go)
+
+deps="
+$top_deps
+$cmd_deps
+"
+
+# Note that grep's exit status is ignored here to allow for packages with no
+# dependencies.
+missing=$(echo "$deps" | \
+	sort | uniq | grep -v '^C$' | \
+	xargs go list -f '{{if not .Standard}}{{join .Deps "\n" }}{{end}}' | sort | uniq | \
```

Comment:
Yeah, it's a little tricky. I'm starting to question whether or not this check is actually providing value commiserate with it's complexity. 

As far as value, It is protecting us against somehow having a green build that happens to close over an unmanaged dependency (and thus lacks reproducibility). A build closing over an unmanaged dep but happening to still succeed seems unlikely, particularly if we do a one-time nuke of the GOPATH on our CIs and don't use `go get`.  And if that did happen, while not great, it seems not catastrophic. Putting those together to get the expected value makes me wonder if maintaining this is really worth the effort.